Extensions/timestamps.js

```diff
@@ -122,23 +121,58 @@ XKit.extensions.timestamps = new Object({
 		}
 
 		this.check_quota();
-		XKit.tools.add_css('#posts .post .post_content { padding-top: 0px; }', "timestamps");
-		if (this.preferences.posts.value || (this.preferences.inbox.value && XKit.interface.where().inbox)) {
-			XKit.post_listener.add("timestamps", this.add_timestamps);
-			this.add_timestamps();
-		}
-
-		if (this.preferences.reblogs.value !== "off") {
-			XKit.post_listener.add("timestamps", this.add_reblog_timestamps);
-			this.add_reblog_timestamps();
-		}
 
 		$(document).on("click", ".xkit-timestamp-failed-why", function() {
 			XKit.window.show("Timestamp loading failed.", "This might be caused by several reasons, such as the post being removed, becoming private, or the Tumblr server having a problem that it can't return the page required by XKit to load you the timestamp.", "error", "<div id=\"xkit-close-message\" class=\"xkit-button\">OK</div></div>");
 		});
 
-		if (this.preferences.only_on_hover.value) {
-			XKit.tools.add_css(" .xtimestamp {display: none; } .post:hover .xtimestamp {display: block; }", "timestamps_on_hover");
+		if (XKit.page.react) {
+			XKit.tools.async_add_function(async () => {
+				/* globals tumblr */
+				return await tumblr.getCssMap();
+			})
+			.then(({post, rightContent, reblog, reblogHeader, badge, blogLink, private}) => {
+				this.posts_class = post.map(x => `.${x}`).join(", ");
+				this.meatballs_class = rightContent.map(x => `.${x}`).join(", ");
+				this.reblogs_class = reblog.map(x => `.${x}`).join(", ");
+				this.reblog_headers_class = reblogHeader.map(x => `.${x}`).join(", ");
+				this.reblog_op_class = badge[1];
+				this.blog_link_class = blogLink.map(x => `.${x}`).join(", ");
+				this.private_posts_class = private.map(x => `.${x}`).join(", ");
+
+				if (this.preferences.posts.value || (this.preferences.inbox.value && XKit.interface.where().inbox)) {
+					this.react_add_timestamps();
+					XKit.post_listener.add("timestamps", this.react_add_timestamps);
+				}
+
+				if (this.preferences.reblogs.value !== "off") {
+					this.react_add_reblog_timestamps();
+					XKit.post_listener.add("timestamps", this.react_add_reblog_timestamps);
+				}
+				
+				if (this.preferences.only_on_hover.value) {
+					XKit.tools.add_css(`.xtimestamp { display: none; } ${this.posts_class.split(", ").map(x => x + ":hover .xtimestamp").join(", ")} { display: block; }`, "timestamps_on_hover");
+				}
+			});
+
+			return;
+
+		} else {
```

Comment:
Agree on the redundancy, but I feel the else block makes it more clear to future programmers that the following is only meant to run on the old Dashboard. Can't resolve by removing that return because it makes the program break. Either or, I'm happy to remove the else block to keep the code consistent of course.

Reply:
huh? how does removing the return break it when there's no other code in the function after the else block?

(also in favour of removing the else block for consistency)

Reply:
I thought it was weird too, XKit wouldn't even load the extension, but I'm unable to reproduce it now so I guess whatever wasn't working is fixed now ¯\_(ツ)_/¯

No worries, removed the else block!